lib/glific/partners.ex

```
@@ -715,15 +717,30 @@ defmodule Glific.Partners do
 
         Goth.Config.add_config(config)
 
-        {:ok, token} =
-          Goth.Token.for_scope(
-            {config["client_email"], "https://www.googleapis.com/auth/cloud-platform"}
-          )
+        Goth.Token.for_scope(
+          {config["client_email"], "https://www.googleapis.com/auth/cloud-platform"}
+        )
+        |> case do
+          {:ok, token} ->
+            token
 
-        token
+          {:error, error} ->
+            Logger.info("Error while fetching token #{error} for org_id #{organization_id}")
+            handle_token_error(organization_id, provider_shortcode, error)
+        end
     end
   end
 
+  @spec handle_token_error(non_neg_integer, String.t(), String.t() | any()) :: nil
+  defp handle_token_error(organization_id, provider_shortcode, error) when is_binary(error) do
+    if String.contains?(error, "account not found"),
+      do: disable_credential(organization_id, provider_shortcode)
+
+    nil
+  end
+
+  defp handle_token_error(_organization_id, _provider_shortcode, _error), do: nil
```

Let's raise an exception for this clause as well.